pkg/executor/sortexec/sort.go

```diff
@@ -180,71 +299,78 @@ func (e *SortExec) fetchRowChunks(ctx context.Context) error {
 	for i, byItem := range e.ByItems {
 		byItemsDesc[i] = byItem.Desc
 	}
-	e.rowChunks = chunk.NewSortedRowContainer(fields, e.MaxChunkSize(), byItemsDesc, e.keyColumns, e.keyCmpFuncs)
-	e.rowChunks.GetMemTracker().AttachTo(e.memTracker)
-	e.rowChunks.GetMemTracker().SetLabel(memory.LabelForRowChunks)
+
+	e.partition = newSortPartition(fields, e.MaxChunkSize(), byItemsDesc, e.keyColumns, e.keyCmpFuncs, e.helper, e.spillLimit)
+	e.partition.getMemTracker().AttachTo(e.memTracker)
+	e.partition.getMemTracker().SetLabel(memory.LabelForRowChunks)
+	e.spillAction = e.partition.actionSpill(e.helper)
+
 	if variable.EnableTmpStorageOnOOM.Load() {
-		e.spillAction = e.rowChunks.ActionSpill()
-		failpoint.Inject("testSortedRowContainerSpill", func(val failpoint.Value) {
-			if val.(bool) {
-				e.spillAction = e.rowChunks.ActionSpillForTest()
-				defer e.spillAction.WaitForTest()
-			}
-		})
+		e.partition.getDiskTracker().AttachTo(e.diskTracker)
+		e.partition.getDiskTracker().SetLabel(memory.LabelForRowChunks)
```

Comment:
why not use `switchToNewSortPartition`?
Why `switchToNewSortPartition` without `if variable.EnableTmpStorageOnOOM.Load()`?

Reply:
> why not use `switchToNewSortPartition`? Why `switchToNewSortPartition` without `if variable.EnableTmpStorageOnOOM.Load()`?

- okk, I will use `switchToNewSortPartition`.
- We call switchToNewSortPartition at the runtime only when the spill has been triggered so it's needless to check that again.